test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoTest.java

```diff
@@ -86,14 +90,55 @@ public void canCrudOneSchedulePlan() {
     
     @Test
     public void getAllSchedulePlans() {
-        TestABSchedulePlan abPlan = new TestABSchedulePlan();
-        TestSimpleSchedulePlan simplePlan = new TestSimpleSchedulePlan();
+        SchedulePlan abPlan = TestABSchedulePlan.create();
+        SchedulePlan simplePlan = TestSimpleSchedulePlan.create();
         
-        schedulePlanDao.createSchedulePlan(abPlan);
-        schedulePlanDao.createSchedulePlan(simplePlan);
+        SchedulePlan plan1 = schedulePlanDao.createSchedulePlan(abPlan);
+        SchedulePlan plan2 = schedulePlanDao.createSchedulePlan(simplePlan);
         
-        List<SchedulePlan> plans = schedulePlanDao.getSchedulePlans(studyIdentifier);
-        assertEquals("2 plans exist", 2, plans.size());
+        List<SchedulePlan> plans = schedulePlanDao.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyIdentifier);
+        assertEquals(getSchedulePlanGuids(plan1, plan2), getSchedulePlanGuids(plans));
+    }
+    
+    @Test
+    public void filtersSchedulePlans() {
+        Set<String> guids = Sets.newHashSet();
+        Set<String> oneGuid = Sets.newHashSet();
+        
+        List<SchedulePlan> plans = TestUtils.getSchedulePlans();
+        SchedulePlan plan = schedulePlanDao.createSchedulePlan(plans.get(0));
+        guids.add(plan.getGuid());
+        
+        plan = schedulePlanDao.createSchedulePlan(plans.get(1));
+        guids.add(plan.getGuid());
+        oneGuid.add(plan.getGuid());
+
+        plan = schedulePlanDao.createSchedulePlan(plans.get(2));
+        guids.add(plan.getGuid());
+        
+        // No known client, all the guids are returned
+        plans = schedulePlanDao.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyIdentifier);
+        assertEquals(guids, getSchedulePlanGuids(plans));
+        
+        // Only one schedule plan matches v9
+        plans = schedulePlanDao.getSchedulePlans(ClientInfo.fromUserAgentCache("app/9"), studyIdentifier);
+        assertEquals(oneGuid, getSchedulePlanGuids(plans));
```

Comment:
I'm going to rewrite this test so that it uses a random study identifier, so it won't conflict with any other tests. I'd like to verify that I get back exactly what I'm expecting and it'll make the test more reliable. No more wiping the tables between runs.